src/cmake-tools.ts

```diff
@@ -2177,24 +2208,21 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
             }
         }
 
-        if (this._lastTerminal !== vscode.env.shell) {
-            void this._launchTerminal?.dispose();
-            this._launchTerminal = undefined;
-        }
-        this._lastTerminal = vscode.env.shell;
-
-        if (!this._launchTerminal) {
-            this._launchTerminal = vscode.window.createTerminal(termOptions);
-        }
+        const terminal = this._createTerminal(termOptions, executable);
 
         let launchArgs = '';
         if (user_config && user_config.args) {
             launchArgs = user_config.args.join(" ");
         }
 
-        this._launchTerminal.sendText(`${executablePath} ${launchArgs}`);
-        this._launchTerminal.show(true);
-        return this._launchTerminal;
+        const quoted = shlex.quote(executable.path);
```

Comment:
It looks like this is a duplicate of line 2201.  We should use `executablePath` instead.